### PR TITLE
ledger support for `__post_init__` fields

### PR DESCRIPTION
**Pull request by EthanMarx**
Adds support to the `ledger` API for derived fields that can be initialized in `__post_init__`. For example, chirp mass, source frame masses, mass ratio etc in  `IntrinsicParameterSet`

### CODE REVIEW

File: libs/analysis/aframe/analysis/ledger/injections.py [hunk 1]
```diff
@@ -35,6 +35,20 @@ class IntrinsicParameterSet(Ledger):
...
+    mass_1_source: np.ndarray = parameter(init=False)
+    mass_2_source: np.ndarray = parameter(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.chirp_mass = (self.mass_1 * self.mass_2) ** (3 / 5) / (
+            self.mass_1 + self.mass_2
+        ) ** (1 / 5)
+
+        self.mass_ratio = self.mass_1 / self.mass_2
```
Comment (wbenoit26) [inline libs/analysis/aframe/analysis/ledger/injections.py:49]: This should be mass2/mass1

File: libs/analysis/aframe/analysis/ledger/injections.py [hunk 2]
```diff
@@ -235,7 +249,8 @@ class InterferometerResponseSet(
     InjectionMetadata, ExtrinsicParameterSet, IntrinsicParameterSet
 ):
     def __post_init__(self):
-        InjectionMetadata.__post_init__(self)
+        # initiate chain of mro __post_init__ calls
```
Comment (wbenoit26) [inline libs/analysis/aframe/analysis/ledger/injections.py:252]: What does "mro" mean?

Comment (EthanMarx) [reply libs/analysis/aframe/analysis/ledger/injections.py:252]: method resolution order. When you have a class that inherits from multiple parents, the `super()` call is ambiguous. So in general, calling `super()` refers to the next class in the method resolution order. In this case the order looks like
```
print(InterferometerResponseSet.__mro__)

(aframe.analysis.ledger.injections.InterferometerResponseSet,
 aframe.analysis.ledger.injections.InjectionMetadata,
 aframe.analysis.ledger.injections.ExtrinsicParameterSet,
 aframe.analysis.ledger.injections.EventParameterSet,
 aframe.analysis.ledger.injections.SkyLocationParameterSet,
 aframe.analysis.ledger.injections.IntrinsicParameterSet,
 aframe.analysis.ledger.ledger.Ledger,
 object)

```
(basically it goes left to right through the inheritance tree). 

So every time super() is called, throughout a class hierarcy, it refers to the next class in line. We can ensure that `Ledgers`  `__post_init__` is eventually called by having super().__post_init__ called in each of these classes

Comment (wbenoit26) [reply libs/analysis/aframe/analysis/ledger/injections.py:252]: Ah neat, okay

**Comment by EthanMarx** [comment]
The problem I found with `properties` is that they aren't picked up by `dataclass` as `fields` (and thus won't be regsitered in `__dataclass_fields__`) so a lot of the existing useful functionality (e.g. aggregating and masking) you wrote will break. I've tested this functionality in the context of the `vizapp`, so mostly calling `.read`, masking, etc. and seems to work smoothly (after some bug fixes such as adding `attr.init == True` checks to a lot of conditionals).

As far as I/O, since all the `init=False` fields are functions of the others they will just be calculated as normal upon initialization. Down the line probably makes sense to save them as well in case a user wants to load in data via `h5py`.

Yeah I think it makes sense for `mass_1` to refer to detector frame, since this is what we use to generate the waveform and inject it into data. 

Will launch a waveform generation run and see how it goes.

**Comment by alecgunny** [comment]
Yeah FWIW I would argue that not getting written to disk is at worst a wash and in some ways a net positive: the `property`s are _functionally_ loaded during `.read` since they can be automatically constructed from the read attributes, with the additional advantage that A) you're not wasting unnecessary disk space, and B) are guaranteed to be consistent, since someone could go into the file and accidentally change the values in that column in a way that makes them disagree with the values they're a function of.

**Pull request closed without merging**